### PR TITLE
Use = default to define empty destructors #2311

### LINKED ISSUE #2311: Use ` = default` to define empty destructors
## Subject of the issue

A handful of types use `{ }` to define empty destructors which is less expressive that using `= default`. Here's a potentially incomplete list of types where I've noticed this is occurring

- `Effect`
- `sf::SoundFileReader`
- `sf::SoundFileWriter`
- `sf::Drawable`
- `sf::Shader`
- `sf::InputStream`
- `sf::WindowImplUIKit`

### PR DESCRIPTION
**Pull request by benjamintli**
Thanks a lot for making a contribution to SFML! 🙂

Before you create the pull request, we ask you to check the follow boxes. (For small changes not everything needs to ticked, but the more the better!)

* [x] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before?
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [ ] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----

## Description

Using `= default` instead of `{}` for empty destructors. Also removed `RenderTextureImpl.cpp` and moved its empty destructor to the header. 

resolves #2311 

## Tasks

* [x] Tested on Linux
* [ ] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

## How to test this PR?

Compiled and ran `window-d` in `examples`

### CODE REVIEW

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

File: src/SFML/Window/iOS/WindowImplUIKit.hpp [hunk 1]
```diff
@@ -72,7 +72,7 @@ class WindowImplUIKit : public WindowImpl
     /// \brief Destructor
     ///
     ////////////////////////////////////////////////////////////
-    ~WindowImplUIKit();
+    ~WindowImplUIKit() = default;
```
Comment (ChrisThrasher) [inline src/SFML/Window/iOS/WindowImplUIKit.hpp:75]: Because this is not virtual, I believe we can remove it entirely

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]

File: src/SFML/Window/iOS/WindowImplUIKit.hpp (hunk 1 above)
Comment (benjamintli) [reply src/SFML/Window/iOS/WindowImplUIKit.hpp]: Fixed

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by Jim-Marsden — APPROVED** [review]

**Review by ChrisThrasher — APPROVED** [review]
Great work, keep it up!

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
Thank you for tackling this!

**Pull request merged**